applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.cpp

```diff
@@ -126,6 +126,14 @@ void CrBeamElement3D2N::GetSecondDerivativesVector(Vector& rValues, int Step)
     KRATOS_CATCH("")
 }
 
+void CrBeamElement3D2N::InitializeNonLinearIteration(ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+    mDeformationPrevious = mDeformationCurrent;

```

Comment:
this should be done by convention in the FinalizeSolutioStep...

Reply:
does not apply here, see other comment
Note that this is not being used like a "historical" variable like e.g. in plasticity, this is reinitialized in each step